scripts/algorithms/KNN.dml

```diff
@@ -0,0 +1,785 @@
+    }
+    #parameter re-define
+    if( select_k==1 ){
+        if(  k_max >= n_records  ){
+            k_max = n_records - 1;
+            print( "k_max should no greater than number of record, change k_max equal " +
+            "( number of record - 1 ) : " + k_max );
+        }
+        if(  k_max >= interval  ){
+            interval = k_max + 1;
+            # k_max should equal interval -1, because we drop self when search nn.
+            print( "interval should be no less than k_max, change interval equal : " + 
+            interval );
+        }
+        if(  k_max <= 1  ){
+            stop( "uncorrect k_max value"  );
+        }   
+        if(  k_min >= k_max ){
+            stop( "k_min >= k_max" );
+        }   
+    }
+    if( select_feature == 1 ){
+        if(  k_value >= n_records  ){
+            k_value = n_records - 1;
+            print( "k_value should be no greater than number of record, change k_value equal " +
+            "( number of record - 1 ) : " + k_value );
+        }
+        #Select feature only
+        StartSelectedPath = ifdef( $START_SELECTED," " );
+        SelectedPath = ifdef( $FEATURE_SELECTED," " );
+        if( StartSelectedPath == " " ){
+            m_start_selected_feature = matrix( 0,1,n_features );
+        }else{
+            m_start_selected_feature = read( StartSelectedPath );
+        }       
+    }
+    CL = read( CLPath );
+    CLType = read( CLTypePath );
+    type = as.integer( as.scalar( CLType[1,1] ) );
+    if( select_k == 1 & select_feature == 1){
+        #Combined k and feature selection
+        print("Start combined k and feature selection ...");
+        [m_selected_feature,s_selected_k] = 
+            getSelectedFeatureAndK( Train,CL,type,m_start_selected_feature,
+                feature_max,k_min,k_max,interval );
+    }else if( select_k == 1 ){
+        #Select k only
+        print("Start k select ...");
+        s_selected_k = getSelectedKBase( Train,CL,type,k_min,k_max,interval );
+    }else if( select_feature == 1 ){
+        #Select feature only
+        print("Start feature selection ... ");
+        [m_selected_feature,d_err] = 
+            getSelectedFeature( Train,CL,type,m_start_selected_feature,
+                feature_max,k_value,interval );
+    }
+}
+if( feature_importance == 1){
+    if(  k_value >= n_records  ){
+        k_value = n_records - 1;
+        print( "k_value should be no greater than number of record, make k_value equal " + 
+        "( number of record - 1 ) : " + k_value );
+    }
+    CL = read( CLPath );
+    CLType = read( CLTypePath );
+    type = as.integer( as.scalar( CLType[1,1] ) );  
+    FeatureImportanceExportPath = ifdef( $FEATURE_IMPORTANCE_VALUE," " );
+    if(FeatureImportanceExportPath != " "){
+        [m_feature_importance,m_norm_feature_importance] = 
+            getFeatureImportance(Train,CL,type,k_value);
+        #print( "feature importance is "  );
+        #tmp = matrixDisplay( m_feature_importance );
```

Comment:
Although commented out but in general it would be good to remove additional prints used for debugging purposes.